src/main/java/ch/hevs/cloudio/endpoint/Transaction.java

```diff
@@ -1,19 +1,18 @@
 package ch.hevs.cloudio.endpoint;
 
-import java.util.LinkedList;
 import java.util.List;
 
-public class Transaction {
-
-    private final List<CloudioAttribute.InternalAttribute> attributes =
-            new LinkedList<CloudioAttribute.InternalAttribute>();
+class Transaction {
+    private final UniqueItemSet<CloudioAttribute.InternalAttribute> attributes = new UniqueItemSet<>();
```

Comment:
@lucblender Using the UniqueItemSet we ensure that every attribute is in the list only once.